internal/internal_workflow_client.go

```diff
@@ -1082,6 +1087,11 @@ func (workflowRun *workflowRunImpl) Get(ctx context.Context, valuePtr interface{
 		return workflowRun.dataConverter.FromPayloads(attributes.Result, valuePtr)
 	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
 		attributes := closeEvent.GetWorkflowExecutionFailedEventAttributes()
+		if attributes.NewExecutionRunId != "" {
+			curRunID := util.PopulatedOnceCell(attributes.NewExecutionRunId)
+			workflowRun.currentRunID = &curRunID
+			return workflowRun.Get(ctx, valuePtr)
+		}
```

Comment:
Can we extract this bit to its own little helper function? Also gives us the opportunity to comment it explaining the reason

Reply:
Sure